packages/stats/gbstats/bayesian/tests.py

```diff
@@ -223,3 +227,114 @@ def get_risk(mu, sigma) -> List[float]:
+        config: BanditConfig,
+    ):
+        self.stat = stat
+        self.config = config
+        self.n_variations = len(stat)
+
+    @property
+    def variation_means(self):
+        return np.array([arm.mean for arm in self.stat])
+
+    @property
+    def variation_variances(self):
+        return np.array([arm.variance for arm in self.stat])
+
+    @property
+    def variation_counts(self):
+        return np.array([arm.n for arm in self.stat])
+
+    @property
+    def prior_precision(self):
+        return np.full(
+            (self.n_variations,),
+            int(self.config.prior_distribution.proper)
+            / self.config.prior_distribution.variance,
+        )
+
+    @property
+    def data_precision(self):
+        return np.array(
+            [
+                float(n) / v if v > 0 else 0
+                for n, v in zip(self.variation_counts, self.variation_variances)
+            ]
+        )
+
+    @property
+    def posterior_precision(self):
+        return self.prior_precision + self.data_precision
+
+    @property
+    def posterior_variance(self):
+        return 1 / self.posterior_precision
+
+    @property
+    def prior_mean(self):
+        return np.full((self.n_variations,), self.config.prior_distribution.mean)
+
+    @property
+    def posterior_mean(self):
+        return self.posterior_variance * (
+            self.prior_precision * self.prior_mean
+            + self.data_precision * self.variation_means
+        )
+
+    # number of Monte Carlo samples to perform when sampling to estimate weights for the SDK
+    @property
+    def n_samples(self):
+        return int(1e4)
+
+    # function that computes thompson sampling variation weights
+    def compute_variation_weights(self):
+        if self.config.iteration < self.config.burn:
+            return np.full((self.n_variations,), 1 / self.n_variations)
+        y = np.random.multivariate_normal(
+            mean=self.posterior_mean,
+            cov=np.diag(self.posterior_variance),
+            size=self.n_samples,
+        )
+        row_maxes = np.max(y, axis=1)
+        p = np.mean((y == row_maxes[:, np.newaxis]), axis=0)
+        if self.config.top_two:
+            return self.top_two_weights(p)
+        return p
+
+    # function that takes weights for largest realization and turns into top two weights
+    @staticmethod
+    def top_two_weights(p):
+        # normalize weights to be no smaller than 1e-5
+        p_star = np.array([max(x, 1e-5) for x in p])
+        p_star /= sum(p_star)
+        n_arms = len(p_star)
+        p_mat = np.repeat(np.expand_dims(p_star, axis=1), n_arms, axis=1)
+        p_mat_t = p_mat.T
+        probs = p_mat * p_mat_t / (1 - p_mat) + p_mat_t * p_mat / (1 - p_mat_t)
+        np.fill_diagonal(probs, 0)
+        return 0.5 * np.sum(probs, axis=1)
+
+    # create config for AB testing from Thompson sampling prior
+    def compute_delta_config(self):
```

Comment:
The goal here is to eventually pass this a series of `EffectBayesianTest` objects to produce effect estimates for the results table?

Reply:
Actually, I was thinking we would output a collection of `EffectBayesianABTest` objects to produce results.